plugins/plugin_utils/fact_diff.py

```diff
@@ -40,46 +43,103 @@ def fact_diff(before, after, plugin):
+            ]
+        if isinstance(before, list):
+            before = "\n".join(map(str, before)) + "\n"
+        if isinstance(after, list):
+            after = "\n".join(map(str, after)) + "\n"
+        return before, after, skip_lines
+
+    def get_fact_diff(self, difflist):
+        if not isinstance(difflist, list):
+            difflist = [difflist]
+        ret = []
+        for diff in difflist:
+            if "before" in diff and "after" in diff:
+                # format complex structures into 'files'
+                for x in ["before", "after"]:
+                    if isinstance(diff[x], MutableMapping):
+                        diff[x] = self._serialize_diff(diff[x])
+                    elif diff[x] is None:
+                        diff[x] = ""
+                if "before_header" in diff:
+                    before_header = "before: %s" % diff["before_header"]
+                else:
+                    before_header = "before"
+                if "after_header" in diff:
+                    after_header = "after: %s" % diff["after_header"]
+                else:
+                    after_header = "after"
+                before_lines = diff["before"].splitlines(True)
+                after_lines = diff["after"].splitlines(True)
+                if before_lines and not before_lines[-1].endswith("\n"):
+                    before_lines[-1] += "\n\\ No newline at end of file\n"
+                if after_lines and not after_lines[-1].endswith("\n"):
+                    after_lines[-1] += "\n\\ No newline at end of file\n"
+                diff_context = (
+                    len(before_lines) if len(before_lines) > len(after_lines) else len(after_lines)
                 )
-                _raise_error(msg)
-
-
-def _xform(before, after, skip_lines):
-    if skip_lines:
-        if isinstance(before, str):
-            before = before.splitlines()
-        if isinstance(after, str):
-            after = after.splitlines()
-        before = [
-            line for line in before if not any(regex.match(str(line)) for regex in skip_lines)
-        ]
-        after = [line for line in after if not any(regex.match(str(line)) for regex in skip_lines)]
-    if isinstance(before, list):
-        before = "\n".join(map(str, before)) + "\n"
-    if isinstance(after, list):
-        after = "\n".join(map(str, after)) + "\n"
-    return before, after, skip_lines
-
-
-def run_diff(before, after, plugin):
-    skip_lines = plugin["vars"].get("skip_lines")
-    _check_valid_regexes(skip_lines=skip_lines)
-    before, after, skip_lines = _xform(before, after, skip_lines=skip_lines)
-    diff = CallbackBase()._get_diff({"before": before, "after": after})
-    ansi_escape = re.compile(r"\x1B[@-_][0-?]*[ -/]*[@-~]")
-    diff_text = ansi_escape.sub("", diff)
-    result = list(diff_text.splitlines())
-    return result
+                differ = difflib.unified_diff(
```

Comment:
@ashwini-mhatre  how about we put a wrapper  for unified_diff in https://github.com/ansible-collections/ansible.utils/blob/main/plugins/module_utils/common/utils.py

Reply:
yes we can add. and then rest of platform also refer that.